src/main/java/com/cloudbees/jenkins/support/SupportPlugin.java

```diff
@@ -668,6 +671,29 @@ public void onOnline(Computer c, TaskListener listener) throws IOException, Inte
                         "Could not install root log handler on node: " + c.getName(), e);
             }
         }
+
+        @Override
+        public void onOffline(@Nonnull Computer c, @CheckForNull OfflineCause cause) {
+            if (Boolean.getBoolean(SupportPlugin.class.getCanonicalName() + ".generateOnOfflineNode")) {
```

Comment:
Indeed, this ought to schedule the task asynchronously, and I agree that waiting a minute would be appropriate. Easy enough to schedule via `Timer.get()`. (I hope `PeriodicWorkImpl` is thread-safe already…)